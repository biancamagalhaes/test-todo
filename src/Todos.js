import React, { useState } from "react";

function Todo(props) {
  const [openList, setOpenList] = useState("none");
  const { todos } = props;
  return (
    <div
      style={{
        width: 270,
        boxShadow: "rgba(50, 50, 50, 0.34) 0px 3px 16px -1px",
        padding: "10px",
        margin: "10px",
      }}
    >
      {todos.image != "" ? (
        <img
          alt="Images"
          style={{ width: "100%", height: 150, backgroundColor: "red" }}
          src={todos.image}
        />
      ) : null}
      <h3 style={{ fontFamily: "sans-serif" }}>{todos.title}</h3>
      <p style={{ fontFamily: "sans-serif" }}>{todos.description}</p>
      {todos.list.length > 0 ? (
        <button
          style={{
            backgroundColor: openList === "none" ? "#2196F3" : "#F44336",
            borderRadius: 5,
            padding: 5,
            border: "none",
            color: "#FFF",
            fontWeight: "bold",
          }}
          onClick={() => setOpenList(openList === "block" ? "none" : "block")}
        >
          {openList === "block" ? "Fechar lista" : "Abrir lista"}
        </button>
      ) : null}
      <ul style={{ display: openList }}>
        {todos.list?.map((todo, index) => (
          <li key={index}>{todo}</li>
        ))}
      </ul>
    </div>
  );
}

export default Todo;
