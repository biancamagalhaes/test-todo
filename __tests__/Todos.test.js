import React from "react";
import { Provider } from "react-redux";
import renderer from "react-test-renderer";
import configureStore from "redux-mock-store";
import Todos from "../src/Todos";

const mockStore = configureStore();

const initialState = {
  todos: [
    { id: 0, text: "Fazer café" },
    { id: 1, text: "Acessar o slack" },
    { id: 2, text: "Beber água" },
  ],
};

const store = mockStore(initialState);

describe("My Connected React-Redux Component", () => {
  const wrapper = renderer
    .create(
      <Provider store={store}>
        <Todos />
      </Provider>
    )
    .toJSON();

  it("should render correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
