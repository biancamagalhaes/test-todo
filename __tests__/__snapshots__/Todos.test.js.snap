// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`My Connected React-Redux Component should render correctly 1`] = `
Array [
  <h2>
    OLA AMIGOS TERRAQUEOS
  </h2>,
  <ul
    className="Todos"
  >
    <li>
      Fazer café
    </li>
    <li>
      Acessar o slack
    </li>
    <li>
      Beber água
    </li>
  </ul>,
]
`;
